File: Coding Escape/telephoneNumbers.py
# By joining the iDroid smartphone development team, you have been given the responsibility of developing the contact manager. Obviously, what you were not told is that there are strong technical constraints for iDroid: the system doesn’t have much memory and the processor is as fast as a Cyrix from the 90s...
class TrieNode:
    def __init__(self):
        self.children = {}

class Trie:
    def __init__(self) :
        self.root = TrieNode()
        self.size = 0

        def insert(self,number):
            current_node = self.root
            for digit in number:
                if digit not in current_node.chidren:
                    current_node.children[digit] = TrieNode()
                    self.size += 1
                current_node = current_node.children[digit]

# Read input
import sys
input = sys.stdin.read
data = input().split()

n=int(data[0])
phone_numbers = data[1:]

#Create a Trie and insert all phone numbers
trie = Trie()
for number in phone_numbers:
    trie.insert(number)

#Output the number of elements in the Trie
print(trie.size)

File: Coding Escape/telephoneNumbers.js
class TrieNode {
    constructor() {
        this.children = {};
        this.count = 0; // to count how many numbers pass through this node
    }
}

class Trie {
    constructor() {
        this.root = new TrieNode();
    }
    
    insert(number) {
        let current = this.root;
        for (let char of number) {
            if (!current.children[char]) {
                current.children[char] = new TrieNode();
            }
            current = current.children[char];
            current.count++;
        }
    }
    
    countUniqueElements() {
        let total = 0;
        function traverse(node) {
            if (!node) return;
            total += node.count;
            for (let child in node.children) {
                traverse(node.children[child]);
            }
        }
        traverse(this.root);
        return total;
    }
}

function countItemsToStore(N, numbers) {
    const trie = new Trie();
    for (let number of numbers) {
        trie.insert(number);
    }
    return trie.countUniqueElements();
}

// Reading input
const N = parseInt(readline());
const numbers = [];
for (let i = 0; i < N; i++) {
    const telephone = readline().trim();
    numbers.push(telephone);
}

// Output the result
console.log(countItemsToStore(N, numbers));
